org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/MavenProjectFacade.java

```diff
@@ -59,6 +74,12 @@
 
   public static final String PROP_CONFIGURATORS = MavenProjectFacade.class.getName() + "/configurators";
 
+  private static final String COMPONENTS_PATH = "META-INF/plexus/components.xml";
+
+  private static final String ARTIFACT_HANDLER_ROLE = "org.apache.maven.artifact.handler.ArtifactHandler";
+
+  private final String[] PACKAGING_TYPES_IGNORING = new String[] {"jar", "war", "ear", "ejb", "pom", "maven-plugin"};
```

Comment:
I think this is not necessary if we make the UI collecting everything in a Set first (see comment below).